### PR TITLE
Implement support for passing custom text resolution for starting log…

### PR DESCRIPTION
**Pull request by kibertoad**
When single application is instantiating multiple instances of fastify, it can become confusing, as it is not clear which one is running on which address. This change allows supplying custom startup text, clarifying log entries.

I am somewhat concerned about this part of the documentation:

```
Both the `options` object and the `callback` parameters follow the [Node.js core](https://nodejs.org/api/net.html#serverlistenoptions-callback) parameter definitions.
```

After this change parameters are still _compatible_ with Node.js parameter definitions, but do not strictly follow them. Not sure if there is a better way to handle this.

#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by jsumners** [comment]
> After this change parameters are still _compatible_ with Node.js parameter definitions, but do not strictly follow them. Not sure if there is a better way to handle this.

The docs need to be updated to basically:

> The options object extends the Node.js options object. Thus, all core options are available with the following additional Fastify specific options:

**Review by jsumners — DISMISSED** [review]
Docs need to be updated.

**Comment by kibertoad** [comment]
@jsumners Done!

**Review by mcollina — APPROVED** [review]
lgtm

**Review by fox1t — APPROVED** [review]

File: test/server.test.js [hunk 1]
```diff
@@ -34,6 +34,16 @@ test('listen should accept stringified number port', t => {
   })
 })
 
+test('listen should accept log text resolution function', t => {
+  t.plan(1)
+
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
+  fastify.listen({ port: '1234', listenTextResolver: () => 'hardcoded text' }, (err) => {
```
Comment (Eomm) [inline test/server.test.js:42]: We should add a `t.pass('executed')` statement here and increment the plan

**Pull request merged**